### DISCUSSION TITLE
TypeScript errors when import useHydrateAtoms and build project

### DISCUSSION THREAD

**Discussion by WhiteCyClop** [pmndrs/jotai#2723] [Bug report] [answered]
### Bug Description

When using `import { useHydrateAtoms } from "jotai/react/utils";`, TS-errors occurs during building project:

![image](https://github.com/user-attachments/assets/63040d38-1358-4990-aa41-a3c92ec5a3a5)

I find PR (#605) that fixed similar problem in the past, maybe this will help

### Reproduction Link

Don't know how to add link to reproduce

**Comment by dai-shi** [comment, marked as answer]
Thanks for reporting. Yeah, we reverted the #605 change afterwards.

Can you add it in your code and see if it works for you?
```ts
declare global {
  interface SymbolConstructor {
    readonly observable: symbol
  }
}
```

**Reply by WhiteCyClop** [reply]
Yes, with that declaration the build has completed successfully

**Reply by WhiteCyClop** [reply]
So this is the solution? No jotai-level fix is ​​needed?

**Reply by dai-shi** [reply]
I think so. Thought #605 was a good fix, but it's not allowed as it's changing the globals. In general, a library shouldn't change global things / make side effects.

**Reply by WhiteCyClop** [reply]
Ok, thank you!
